src/blogs.rs

```diff
@@ -255,15 +259,16 @@ decl_module! {
 
     fn update_blog(origin, blog_id: T::BlogId, update: BlogUpdate<T>) {
       let owner = ensure_signed(origin)?;
-      
-      let has_updates = 
+
+      let has_updates =
         update.writers.is_some() ||
         update.slug.is_some() ||
         update.json.is_some();
 
       ensure!(has_updates, "Nothing to update in a blog");
 
       let mut blog = Self::blog_by_id(blog_id).ok_or("Blog was not found by id")?;
+      let mut fields_changed = 0;
```

Comment:
Let's use the same approach with `fields_changed` in `update_post` function as well